src/arithimpl/gmpimpl.rs

```diff
@@ -45,7 +55,9 @@ impl ModularArithmetic for Mpz {
         a.gcdext(b)
     }
 
-    // TODO: native way of doing divmod (supported by GMP but not currently by Rust wrapper)
+    fn divmod(dividend: &Self, module: &Self) -> (Self, Self) {
+        (dividend.div(module), dividend.rem(module))
```

Comment:
is this better than the default implementation? otherwise it might be clearer to keep the TODO in case rust-gmp is ever extended